---
permalink: /
title: "Datasets"
author_profile: true
---

## Machine learning benchmarks
Microbiome benchmarks for machine learning from the [Vangay et al MLRepo paper](https://academic.oup.com/gigascience/article/8/5/giz042/5481665) are [here](https://github.com/knights-lab/MLRepo).

## Daily diet-microbiome variation study
Processed shotgun metagenomics data, dietary intake data, and all metadata from the Johnson et al 2019 Cell Host and Microbe paper, "Daily Sampling Reveals Personalized Diet-Microbiome Associations" are available [here](https://github.com/knights-lab/dietstudy_analyses). Related FoodTree tree and code are [here](https://github.com/knights-lab/Food_Tree).

## Immigration Microbiome Project
Processed 16S microbiome data, shotgun metagenomics data, dietary intake data, and all metadata from the Vangay et al 2018 Cell paper, "U.S. Immigration Westernizes the Human Microbiome" are available [here](https://github.com/knights-lab/IMP_analyses).

## FEMS Benchmarks
Raw and processed benchmark data, including metadata, from the Knights 2011 FEMS paper, "Supervised Classification of Human Microbiota" are available [here](https://drive.google.com/drive/folders/0B_29m5gMd9XBOHV3OFFCcTNLcVk?resourcekey=0-vCBl1xIDlkIpkkJm-EdRKw&usp=sharing).

## SourceTracker
Data used in SourceTracker publication, Knights et al. Nature Methods 2011, "Bayesian Community-Wide Culture-Independent Microbial Source Tracking" are available [here](https://drive.google.com/drive/folders/0B6AdcLiMuqsCMXBhLXhzVmNxQ2M?resourcekey=0-HpJ_VtGEddvnjmqgwGMgUg&usp=sharing).

## Enterotypes
Final data used in Koren et al. 2013, "A Guide to Enterotypes accross the Human Body: Meta-Analysis of Microbial Community Structures in Human Microbiome Datasets" are [here](https://drive.google.com/file/d/0B6AdcLiMuqsCUTNxZkJ0YzdDNE0/view?usp=sharing&resourcekey=0-7m3NKJ7eM0e2YcewVMulSg).

## Host Genetics and the Microbiome in IBD
Final data analyzed in Knights et al. 2014, "Complex Host Genetics Influence the Microbiome in Inflammatory Bowel Disease" are [here](https://drive.google.com/drive/folders/0B6AdcLiMuqsCfkJIdFNTdGJMXzhxRGZxMEZhNnBIbHpvbUhfNl9zX24yTXlNMERSN0dsRzQ?resourcekey=0-hF9x4ETLJdYWJSYLJC-a1A&usp=sharing).

## Predicting Chemotherapy-related Infection
Publication data used in Montassier et al. 2016, "Pretreatment Gut Microbiome Predicts Chemotherapy-related Infection" are [here](https://drive.google.com/drive/folders/0B6AdcLiMuqsCLWItLUMtSFFCcHc?resourcekey=0-kvEIwI3Pf574iRvb4ddDGA&usp=sharing).

## Captivity Humanizes the Primate Microbiome
Species and lifestyle metadata and OTU tables used in Clayton et al. 2016, "Captivity Humanizes the Primate Microbiome" are [here](https://drive.google.com/drive/folders/0B6AdcLiMuqsCZVZpaFF3STRlN2M?resourcekey=0-gVrmv_StuOyXpYO9AELKiw&usp=sharing)https://drive.google.com/drive/folders/0B6AdcLiMuqsCZVZpaFF3STRlN2M?resourcekey=0-gVrmv_StuOyXpYO9AELKiw&usp=sharing. Raw sequencing data [here at EBI under project PRJEB11414](http://www.ebi.ac.uk/ena/data/view/PRJEB11414)http://www.ebi.ac.uk/ena/data/view/PRJEB11414.
